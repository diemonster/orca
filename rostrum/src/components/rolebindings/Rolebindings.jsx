import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import RolebindingList from './RolebindingList';

export function Rolebindings(props) {
  const { selectedNamespace } = props;
  if (selectedNamespace !== '') {
    return (
      <RolebindingList />
    );
  }

  return (
    <div className="rolebinding-container">
      <h3>no namespace selected</h3>
    </div>
  );
}

Rolebindings.propTypes = {
  selectedNamespace: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  selectedNamespace: state.namespace.selectedNamespace,
});

export default connect(mapStateToProps)(Rolebindings);
