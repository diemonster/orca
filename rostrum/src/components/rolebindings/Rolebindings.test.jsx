import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Rolebindings } from './Rolebindings';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('Rolebindings', () => {
    it('should render itself when no namespace is selected', () => {
      const props = {
        selectedNamespace: '',
      };

      const rolebindings = shallow(<Rolebindings {...props} />);
      expect(rolebindings).toMatchSnapshot();
    });

    it('should render itself and subcomponents when a namespace is selected', () => {
      const props = {
        selectedNamespace: 'some-namespace',
      };

      const rolebindings = shallow(<Rolebindings {...props} />);
      expect(rolebindings).toMatchSnapshot();
    });
  });
});
