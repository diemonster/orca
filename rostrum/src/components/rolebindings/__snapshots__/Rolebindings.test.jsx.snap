// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Rolebindings should render itself and subcomponents when a namespace is selected 1`] = `<Connect(RolebindingList) />`;

exports[`components Rolebindings should render itself when no namespace is selected 1`] = `
<div
  className="rolebinding-container"
>
  <h3>
    no namespace selected
  </h3>
</div>
`;
